tests/Library/Core/ArrayTranslateTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * @author Patrick Kelly <patrick.k@vanillaforums.com>
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license Proprietary
+ */
+
+namespace VanillaTests\Library\Core;
+
+use VanillaTests\SharedBootstrapTestCase;
+
+/**
+ * Tests for arrayTranslate().
+ */
+class ArrayTranslateTest extends SharedBootstrapTestCase {
+
+    /**
+     * Test  {@link arrayTranslate()} against several scenarios.
+     *
+     * @param array $testArray Array to be translated.
+     * @param array $testMap The map of what the array keys will be.
+     * @param array $addRemaining Add keys from the array map to the array even if they do not exist on the array.
+     * @param array $expected
+     * @dataProvider provideTranslateableArrays
+     */
+    public function testArrayTranslate($testArray, $testMap, $addRemaining, $expected) {
+        $value = arrayTranslate($testArray, $testMap, $addRemaining[0]);
+        $this->assertSame($expected, $value);
+    }
+
+    /**
+     * Provide test data for {@link arrayTranslate()}.
+     *
+     * @return array Returns an array of test data.
+     */
+    public function provideTranslateableArrays() {
+        $r = [
+            'simpleAssocAdd' => [['badKey' => 'a.b'], ['badKey' => 'fixedKey'], [0 => 1], ['fixedKey' => 'a.b']],
```

Comment:
If possible, could you break up the the parameters into separate lines? I find it can make parsing these test cases a lot easier to read.

Some of these lines are particularly long.